src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -90,19 +113,27 @@ private void HandleBinaryOperator(in OperationAnalysisContext context, INamedTyp
             context.ReportDiagnostic(binary.Syntax.GetLocation().CreateDiagnostic(Rule));
         }
 
-        private static bool IsSymbolType(IOperation operation, INamedTypeSymbol symbolType)
+        private static void HandleMethodReferenceOperation(OperationAnalysisContext context, INamedTypeSymbol symbolType)
```

Comment:
Are you sure this callback is being made in presence of invocations? We have https://github.com/dotnet/roslyn/issues/26206 which tracks that `IInvocationOperation` does not expose the underlying `IMethodReferenceOperation` in the operation tree. If you are getting this callback for invocations, then that bug might be fixed, otherwise you would need to registered for `OperationKind.Invocation` instead.

Reply:
I thought I was, but it's been a bit since I ran this. Testing again now...

Reply:
Changed to use IInvocationOperation